core/src/test/scala/org/apache/spark/sql/delta/DeltaErrorsSuite.scala

```diff
@@ -143,7 +143,8 @@ trait DeltaErrorsSuiteBase
     testUrls()
   }
 
-  test("test DeltaErrors methods") {
+  // Split to multiple tests to avoid JDK issue if running on JDK 17.
+  test("test DeltaErrors methods part 1") {
```

Comment:
Split this test to multiple tests to avoid JDK issue if running on JDK 17:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (metaspaceArena.cpp:93), pid=45644, tid=6147
#  guarantee(requested_word_size <= chunklevel::MAX_CHUNK_WORD_SIZE) failed: Requested size too large (528698) - max allowed size per allocation is 524288.
#
# JRE version: Java(TM) SE Runtime Environment (17.0+35) (build 17+35-LTS-2724)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (17+35-LTS-2724, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-amd64)
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/yumwang/opensource/delta/core/hs_err_pid45644.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#
Exception in thread "Thread-9" java.io.EOFException
  | => cat java.base/java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:3192)
        at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1693)
        at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:514)
        at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:472)
        at org.scalatest.tools.Framework$ScalaTestRunner$Skeleton$1$React.react(Framework.scala:839)
        at org.scalatest.tools.Framework$ScalaTestRunner$Skeleton$1.run(Framework.scala:828)
        at java.base/java.lang.Thread.run(Thread.java:833)

```

Reply:
How to reproduce:
```shell
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home
export PATH=${JAVA_HOME}/bin:${PATH}

https://github.com/delta-io/delta.git
cd delta
git checkout 6a30e958de4322100b2ccfa13fa29ae155369a07
build/sbt clean  "core/testOnly *.DeltaErrorsSuite"
```

Reply:
Is it a java 17 bug?

Reply:
I'm not sure. but I have report a bug to https://bugreport.java.com/bugreport/crash.jsp.

Reply:
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8294677 cc @rednaxelafx

Reply:
Thanks for the ping, @wangyum !
Do you happen to have the `/Users/yumwang/opensource/delta/core/hs_err_pid45644.log` file (or other instances of the hs_err file from the same kind of run)? Could you please share it for more information? Thanks!

Reply:
Yes. [hs_err_pid45644.log](https://github.com/delta-io/delta/files/9702090/hs_err_pid45644.log). You can also reproduce it with:
```sh
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home
export PATH=${JAVA_HOME}/bin:${PATH}

git clone https://github.com/delta-io/delta.git
cd delta
git checkout 6a30e958de4322100b2ccfa13fa29ae155369a07
build/sbt clean  "core/testOnly *.DeltaErrorsSuite"
```